gitlab/client.py

```diff
@@ -1110,6 +1111,8 @@ def _query(
                 next_url = requests.utils.parse_header_links(result.headers["links"])[
                     0
                 ]["url"]
+            if not next_url.startswith(self._gl._base_url):
```

Comment:
Since I'm not as smart as most people, I would appreciate a comment which explains why this is here and what it is doing.